### PR TITLE
[feature][no, really] APIv2 URL helper functions

### PR DESCRIPTION
**Pull request by abought**
Restores changes accidentally merged in #98 and reverted in #99, and adds JS unit tests. Also fixes bug whwere JS function could accidentally include doubled slashes.

For full description, see the original submission in centerforopenscience/osf.io/#2895

### CODE REVIEW

File: website/static/js/osfHelpers.js [hunk 1]
```diff
@@ -23,6 +23,41 @@ var growl = function(title, message, type) {
...
+    apiPrefix = apiPrefix || window.contextVars.apiV2Prefix;
+
+    // Don't output double slashes when concatenating two strings with adjoining slashes
+    if (apiPrefix && pathString && apiPrefix.charAt(apiPrefix.length - 1) === "/" && pathString.charAt(0) === "/"){
+        pathString = pathString.substring(1); // Strip off the redundant leading slash
+    }
+
+    var apiUrl = apiPrefix + pathString;
+    // Add parameters to URL (if any). Ensure encoding as necessary
+    if (paramsObject){
```
Comment (brianjgeiger) [inline website/static/js/osfHelpers.js]: $.param() may save you some trouble here.

**Comment by brianjgeiger** [comment]
Just the one thing in this pass. :penguin:

File: website/static/js/osfHelpers.js (hunk 1 above)
Comment (abought) [inline website/static/js/osfHelpers.js]: That was a pleasant surprise. Thanks!

**Comment by brianjgeiger** [comment]
Okay, looks good. :tada: :penguin: :tada: 

Steve may end up with some requests on the docstrings, but I'm not positive, so we'll start with this and make changes if he wants.

**Pull request merged**